### PR TITLE
Moved scene explorer logic to its own function

### PR DESCRIPTION
**Pull request by OsiJr**

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -1272,6 +1272,138 @@ void vcRenderScene_HandlePicking(vcState *pProgramState, vcRenderData &renderDat
...
+
+  if (vcMenuBarButton(pProgramState->pUITexture, vcString::Get("sceneExplorerAddUDS"), buffer, vcMBBI_AddPointCloud, vcMBBG_FirstItem) || vcHotkey::IsPressed(vcB_AddUDS))
+    vcModals_OpenModal(pProgramState, vcMT_AddSceneItem);
+
+  if (vcMenuBarButton(pProgramState->pUITexture, vcString::Get("sceneExplorerAddFolder"), nullptr, vcMBBI_AddFolder, vcMBBG_SameGroup))
+  {
+    vdkProjectNode *pNode = nullptr;
+    if (vdkProjectNode_Create(pProgramState->activeProject.pProject, &pNode, pProgramState->activeProject.pRoot, "Folder", vcString::Get("sceneExplorerFolderDefaultName"), nullptr, nullptr) != vE_Success)
+    {
+      vcState::ErrorItem projectError;
```
Comment (peterAdams1014) [inline src/main.cpp]: vcState::ErrorItem projectError = {};

File: src/main.cpp [hunk 2]
```diff
@@ -1272,6 +1272,138 @@ void vcRenderScene_HandlePicking(vcState *pProgramState, vcRenderData &renderDat
...
+
+      pProgramState->errorItems.PushBack(projectError);
+
+      vcModals_OpenModal(pProgramState, vcMT_ProjectChange);
+    }
+  }
+
+  if (vcMenuBarButton(pProgramState->pUITexture, vcString::Get("sceneExplorerAddViewpoint"), nullptr, vcMBBI_SaveViewport, vcMBBG_SameGroup))
+  {
+    vdkProjectNode *pNode;
```
Comment (peterAdams1014) [inline src/main.cpp]: vdkProjectNode *pNode = nullptr;

File: src/main.cpp [hunk 3]
```diff
@@ -1272,6 +1272,138 @@ void vcRenderScene_HandlePicking(vcState *pProgramState, vcRenderData &renderDat
...
+      if (pProgramState->gis.isProjected)
+        vdkProjectNode_SetGeometry(pProgramState->activeProject.pProject, pNode, vdkPGT_Point, 1, &cameraPositionInLongLat.x);
+
+      vdkProjectNode_SetMetadataDouble(pNode, "transform.rotation.x", pProgramState->camera.eulerRotation.x);
+      vdkProjectNode_SetMetadataDouble(pNode, "transform.rotation.y", pProgramState->camera.eulerRotation.y);
+      vdkProjectNode_SetMetadataDouble(pNode, "transform.rotation.z", pProgramState->camera.eulerRotation.z);
+    }
+    else
+    {
+      vcState::ErrorItem projectError;
```
Comment (peterAdams1014) [inline src/main.cpp]: vcState::ErrorItem projectError = {};

**Comment by OsiJr** [comment]
@peterAdams1014 , fixed those things- this is just moved code though which we normally just push through?

**Pull request merged**